### ISSUE TITLE
Bias mitigation and fairness boosting  in multi class classification use case using Amazon SageMaker Clarify

### ISSUE THREAD

**Issue by pradeepdev-1995** [aws/amazon-sagemaker-examples#3779]
Is Amazon SageMaker Clarify support mitigating bias and improving fairness in multi class classification problems?
If so please provide an example for doing the same in a sample multi class classification dataset.

**Comment by xgchena** [comment]
Thank you for reaching out to us.

In this context I guess you want to use SageMaker Clarify processing job for bias analysis. Currently the job supports detecting potential bias during data preparation (pre-training) and in trained model (post-training), but it does not support bias mitigation methods yet.

To detect bias for multi-class classification problems: when you configure the Clarify job for bias analysis, there is a parameter called `label_values_or_threshold`. If the label is multi-class, the parameter should be a list of label values that are favorable to a demographic group observed in a sample.

E.g., if the label column of your dataset has three classes "A", "B" and "C", and samples that have label value "A" or "B" designates positive result, then assign `["A", "B"]` to the parameter,

* If you use the [SageMakerClarifyProcessor](https://sagemaker.readthedocs.io/en/stable/api/training/processing.html#sagemaker.clarify.SageMakerClarifyProcessor) API in SageMaker Python SDK,

  ```
  import sagemaker
  bias_config = sagemaker.clarify.BiasConfig(
      label_values_or_threshold=["A", "B"],
      ...
  )
  ```

* If you compose the [analysis configuration file](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-processing-job-configure-analysis.html) by yourself, the file should have this entry,

  ```
  {
      ...
      label_values_or_threshold: ["A", "B"],
      ...
  }
  ```

FYI: SageMaker Clarify uses [these terms](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-detect-data-bias.html) to discuss bias and fairness, and here are the definition of [pretraining bias metrics](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-detect-data-bias.html) and [posttraining bias metrics](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-measure-post-training-bias.html).
 
If you have any additional questions regarding the topic, we suggest that you post your queries to the AWS re:Post forum. It is a great platform to get support from the community and our experts. Please remember to include the tag [Amazon SageMaker Clarify](https://repost.aws/tags/TAlEAotjPdR02YRB_rZdkGng/amazon-sage-maker-clarify) in your post to help us better understand your query and provide you with the most relevant assistance. We hope this helps and look forward to hearing from you on the forum.
